test/trees/prufer.jl

```diff
@@ -0,0 +1,61 @@
+function _harmonize_type(g::AbstractGraph{T}, c::Vector{S}) where {T,S<:Integer}
+    return convert(Vector{T}, c)
+end
+
+@testset "Tree utilities" begin
+    t1 = Graph(6)
+    for e in [(1, 4), (2, 4), (3, 4), (4, 5), (5, 6)]
+        add_edge!(t1, e...)
+    end
+    code = [4, 4, 4, 5]
+
+    t2 = path_graph(2)
+    t3 = star_graph(10)
+    t4 = binary_tree(3)
+
+    g1 = cycle_graph(8)
+    g2 = complete_graph(4)
+    g3 = Graph(5)
+
+    d1 = cycle_digraph(5)
+    d2 = path_digraph(5)
+    d3 = DiGraph(2)
+    add_edge!(d3, 1, 2)
+    add_edge!(d3, 2, 1)
+
+    @testset "tree_check" begin
+        for t in testgraphs(t1, t2, t3)
+            @test is_tree(t)
+        end
+        for g in testgraphs(g1, g2, g3)
+            @test !is_tree(g)
+        end
+        for g in testgraphs(d1, d2, d3)
+            @test_throws MethodError is_tree(g)
+        end
+    end
```

Comment:
Can we also have the following special test cases here:
- An empty graph
- A graph with a single vertex
- Two single vertices
- A graph with a self-loop
- And maybe a graph that is a forest of two trees

I made the experience, that our algorithms often error on such test cases.